build/pipeline-perf-tests-automation.yaml

```diff
@@ -0,0 +1,46 @@
+# Build for performance tests
+trigger:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - src/client/Microsoft.Identity.Client/*
+
+pr: none
+
+pool:
+  name: MsalNetPerfTest1ESPool
+  demands: msbuild
+
+variables:
+  BuildPlatform: 'any cpu'
+  BuildConfiguration: 'release'
+  MSAL_DESKTOP_ONLY_DEV: 'false' # skips building mobile and legacy targets
+
+jobs:
+  - job: 'BuildJob'
+    displayName: 'Build and run tests'
+    steps:
+    - task: NuGetToolInstaller@1
+      displayName: 'Use NuGet 5.x'
+      inputs:
+        versionSpec: 5.x
+
+    - task: NuGetCommand@2
+      displayName: 'NuGet restore'
+      inputs:
+        restoreSolution: PerformanceTests.sln
+
+    - task: MSBuild@1
+      displayName: Build PerformanceTests.sln
+      inputs:
+        solution: PerformanceTests.sln
+        msbuildArchitecture: x64
+        platform: '$(BuildPlatform)'
+        configuration: '$(BuildConfiguration)'
+        msbuildArguments: '/p:RunCodeAnalysis=false /p:MSAL_DESKTOP_ONLY_DEV=$(MSAL_DESKTOP_ONLY_DEV)'
+        clean: true
+
+    - script: '$(Build.SourcesDirectory)\tests\Microsoft.Identity.Test.Performance\bin\$(BuildConfiguration)\$(BuildRuntime)\Microsoft.Identity.Test.Performance.exe -BranchName $(Build.SourceBranch) -BranchCommit $(Build.SourceVersion)'
```

Comment:
I have a few questions about this, perhaps let's discuss at standup.

- how are the 2 params used?
- how are the results reported / used? Or will this be part of another PR?

Reply:
Currently the results are printed to the console window in the build pipeline and those parameters are not used. But I added them because I was curious if it's possible to pass the branch info into the perf app, and it is. So in the future the branch info can be stored along with the perf run results.

Reply:
Ack, maybe add a comment in the code or in the yaml about this.